tests/test_api.py

```diff
@@ -187,3 +189,81 @@ def test_ask_analysis_result(self):
         data = {"job_id": job_id}
         response = self.client.get("/api/ask_analysis_result", data)
         self.assertEqual(response.status_code, 200)
+
+    def test_list_all_jobs(self):
+        response = self.client.get("/api/jobs")
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_job_by_id_no_admin(self):
+        job_id = os.environ.get("TEST_JOB_ID", "1")
+        response = self.client.get(f"/api/jobs/{job_id}")
+        self.assertEqual(response.status_code, 404)
+
+    def test_get_job_by_id_admin(self):
+        assign_perm("api_app.view_job", self.user)
```

Comment:
i think this should be similar to Line 238.

Reply:
I'm confused about this. Previously also I thought about this, but we are not creating any job in the `setUp` of `ApiJobTests` class so why is the test `test_ask_analysis_result` successful when it sends a job id in the `data` field - asking the result for a job that has never been created? Assuming `setUp` runs everytime before each test and creates a new test database.

Reply:
nice catch! turns out it's unintended behavour since we have hardcoded 200 status code in the `ask_analysis_resutlt` function. (https://github.com/intelowlproject/IntelOwl/blob/7463881edc8448fd6f9c78a8ca192412c1a49578/api_app/api.py#L346)

Reply:
Oh that makes sense. Looking at the `api.py` file, this seems to be a common theme followed in other endpoints as well so we could `assert` the response data in this test instead of the reponse code?

> i think this should be similar to Line 238.
Yes but I wasn't sure what kind of a test job I should create since the model has a lot of parameters - what kind of a sample job I should create, even thought it might make much of a difference for testing purposes.